streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -584,6 +579,7 @@ public boolean test(final K1 key, final V1 value) {
         String name = topology.newName(LEFTJOIN_NAME);
 
         topology.addProcessor(name, new KStreamKTableLeftJoin<>((KTableImpl<K, ?, V1>) other, joiner), this.name);
+        topology.connectProcessorAndStateStores(name, other.getStoreName());
```

Comment:
My bad. I added a test for it, which did not do what I expected, so I though the problem is fixed. Now I added a proper integration test and also did really fix the issue (at least I hope so).